app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -1,62 +1,137 @@
+const genBackgroundImage = () => `url(${Math.random() > 0.5 ? dataUrl1 : dataUrl2})`
 
-  return (
-    <div className="bg-white">
-      <div className="mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:px-8">
-        <div className="grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 xl:gap-x-8">
-          {templates.map((template, index) => (
-            <div
-              key={index}
-              className="h-40 rounded-lg border-4 border-dashed border-gray-200 flex"
-            >
-              <div className="m-auto text-center">
-                <button
-                  onClick={() => {
-                    onChange(template.templateName);
-                  }}
+/** Render single template */
+interface TemplateRenderProps extends TemplatesProps {
+    template: Template
+}
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const TemplateRender: React.FC<TemplateRenderProps> = ({ template, onChange }) => {
```

Comment:
i'm not sure `TemplateRender` is a good name - especially if the other component is just called `Templates`

it *would* be the same name as the interface at the top of the file, but:
- it's not actually a typescript error to have a type and variable with the same name
- you may want to consider renaming the interface at the top instead - to something like `TemplateData` or `TemplateInfo`

Reply:
Yeah, the `Templates` have already been defined in the type space, so I'm trying to avoid confusion by not renaming them.
I think using `TemplatesRender(templates, onChange)` to render the `templates` array would be a good approach.
What do you think?